Crash when back button pressed if NearbyList has not finished loading

```
USER_COMMENT=gallery button then back
ANDROID_VERSION=6.0.1
APP_VERSION_NAME=1.30


BRAND=google
PHONE_MODEL=Nexus 5
CUSTOM_DATA=
STACK_TRACE=java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.content.Context.getSystemService(java.lang.String)' on a null object reference
at android.view.LayoutInflater.from(LayoutInflater.java:229)
at android.widget.ArrayAdapter.<init>(ArrayAdapter.java:178)
at android.widget.ArrayAdapter.<init>(ArrayAdapter.java:163)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAdapter.<init>(NearbyListFragment.java:178)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.onPostExecute(NearbyListFragment.java:143)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.onPostExecute(NearbyListFragment.java:112)

at android.os.AsyncTask.finish(AsyncTask.java:651)
at android.os.AsyncTask.-wrap1(AsyncTask.java)
at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:668)
at android.os.Handler.dispatchMessage(Handler.java:102)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5461)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

```
USER_COMMENT=trying to post a picture of the weather in Nigeria
ANDROID_VERSION=5.1
APP_VERSION_NAME=1.31
BRAND=itel
PHONE_MODEL=itel it1506
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: An error occured while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:304)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
at java.util.concurrent.FutureTask.run(FutureTask.java:242)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
at java.util.ArrayList.get(ArrayList.java:308)
at fr.free.nrw.commons.nearby.NearbyListFragment.loadAttractionsFromLocation(NearbyListFragment.java:308)
at fr.free.nrw.commons.nearby.NearbyListFragment.access$300(NearbyListFragment.java:27)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.doInBackground(NearbyListFragment.java:134)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.doInBackground(NearbyListFragment.java:112)
at android.os.AsyncTask$2.call(AsyncTask.java:292)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
... 4 more
java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0
at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
at java.util.ArrayList.get(ArrayList.java:308)
at fr.free.nrw.commons.nearby.NearbyListFragment.loadAttractionsFromLocation(NearbyListFragment.java:308)
at fr.free.nrw.commons.nearby.NearbyListFragment.access$300(NearbyListFragment.java:27)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.doInBackground(NearbyListFragment.java:134)
at fr.free.nrw.commons.nearby.NearbyListFragment$NearbyAsyncTask.doInBackground(NearbyListFragment.java:112)
at android.os.AsyncTask$2.call(AsyncTask.java:292)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
at java.lang.Thread.run(Thread.java:818)
```

--- a/app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java
@@ -109,6 +109,15 @@ public class NearbyListFragment extends ListFragment implements TaskListener {
         super.onDetach();
     }
 
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+
+        if(nearbyAsyncTask != null && nearbyAsyncTask.getStatus() != AsyncTask.Status.FINISHED) {
+            nearbyAsyncTask.cancel(true);
+        }
+    }
+
     private class NearbyAsyncTask extends AsyncTask<Void, Integer, List<Place>> {
 
         private final TaskListener listener;
@@ -138,6 +147,11 @@ public class NearbyListFragment extends ListFragment implements TaskListener {
         @Override
         protected void onPostExecute(List<Place> result) {
             super.onPostExecute(result);
+
+            if(isCancelled()) {
+                return; 
+            }
+
             progressBar.setVisibility(View.GONE);
 
             mAdapter = new NearbyAdapter(getActivity(), places);

--- a/app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java
+++ b/app/src/main/java/fr/free/nrw/commons/nearby/NearbyListFragment.java
@@ -149,7 +149,7 @@ public class NearbyListFragment extends ListFragment implements TaskListener {
             super.onPostExecute(result);
 
             if(isCancelled()) {
-                return; 
+                return;
             }
 
             progressBar.setVisibility(View.GONE);
@@ -318,10 +318,12 @@ public class NearbyListFragment extends ListFragment implements TaskListener {
             );
         }
 
-        for(int i = 0; i < 500; i++) {
-            Place place = places.get(i);
-            String distance = formatDistanceBetween(mLatestLocation, place.location);
-            place.setDistance(distance);
+        if (places.size() > 0) {
+            for (int i = 0; i < 100; i++) {
+                Place place = places.get(i);
+                String distance = formatDistanceBetween(mLatestLocation, place.location);
+                place.setDistance(distance);
+            }
         }
         return places;
     }

--- a/app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java
+++ b/app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java
@@ -26,6 +26,7 @@ public class NearbyPlaces {
                 StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
                 StrictMode.setThreadPolicy(policy);
 
+
                 URL file = new URL("https://tools.wmflabs.org/wiki-needs-pictures/data/data.csv");
 
                 BufferedReader in = new BufferedReader(new InputStreamReader(file.openStream()));

